OpenRA.Mods.Common/Activities/DeliverUnit.cs

```diff
@@ -41,6 +41,9 @@ public DeliverUnit(Actor self, Target destination, WDist deliverRange)
 
 		protected override void OnFirstRun(Actor self)
 		{
+			if (self.Trait<Carryall>().Carryable == null)
```

Comment:
The `Carryall` reference is already cached as `carryall`. Please use this instead of querying it again!

Reply:
Can you please also add a comment here explaining about `AutoCarryall`?